src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -181,6 +197,17 @@ public Scheduler()
                 }
             }
 
+            if (!int.TryParse(Environment.GetEnvironmentVariable("MSBUILDCORELIMIT"), out _coreLimit) || _coreLimit <= 0)
+            {
+                _coreLimit = NativeMethodsShared.GetLogicalCoreCount();
+            }
+            if (!int.TryParse(Environment.GetEnvironmentVariable("MSBUILDNODECOREALLOCATIONWEIGHT"), out _nodeCoreAllocationWeight)
+                || _nodeCoreAllocationWeight <= 0
+                || _nodeCoreAllocationWeight > 100)
+            {
+                _nodeCoreAllocationWeight = 0;
+            }
```

Comment:
I know you said you were testing this out, but this is probably the most confusing part of this PR from a user's perspective to me.

Reply:
I have added more comments in this section, apologies for the terse code.